plugins/modules/system_high_availability_settings.py

```diff
@@ -0,0 +1,425 @@
+                    echo $key.','.$item['description'].'\n';
+                }
+                """
+
+    # run php function
+    result = opnsense_utils.run_command(
+        php_requirements=php_requirements,
+        command=php_command,
+    )
+
+    # check for stderr
+    if result.get("stderr"):
+        raise OPNSenseGetInterfacesError("error encountered while getting services")
+    allowed_services = dict(
+        service.split(",") for service in result.get("stdout_lines")
+    )
+    return allowed_services
+
+
+def services_to_synchronize(config: OPNsenseModuleConfig, sync_services: List[str]):
+    """
+    Handler function for the setting services_to_synchronize.
+    Args:
+        config (OPNsenseModuleConfig): The configuration for the opnsense firewall
+        sync_services (List[str]): A list of services that should be synchronized
+    """
+    allowed_services = plugins_xmlrpc_sync()
+    if isinstance(sync_services, str):
+        sync_services = [sync_services]
+    for service in sync_services:
+        if service in allowed_services.keys():
+            service_id = service
+            service_description = allowed_services[service]
+        elif service in allowed_services.values():
+            service_id = {v: k for k, v in allowed_services.items()}[service]
+            service_description = service
+        else:
+            raise ValueError(
+                f"Service {service} could not be found in your Opnsense installation. "
+                + f"These are all the available services: {', '.join(allowed_services.values())}."
+            )
+
+        service_xml_path = f"synchronize{service_id}"
+        if config.get("hasync").find(service_xml_path) is None:
+            xml_elem = Element(service_xml_path)
+            xml_elem.text = "on"
+            config.get("hasync").append(xml_elem)
+    for service_id, service_description in allowed_services.items():
+        service_xml_elem = config.get("hasync").find(f"synchronize{service_id}")
+        if (
+            service_id not in sync_services
+            and service_description not in sync_services
+            and service_xml_elem is not None
+        ):
+            config.get("hasync").remove(service_xml_elem)
+
+
+def main():
+    """
+    Main function of the system_high_availability_settings module
+    """
+
+    module_args = {
+        "synchronize_states": {"type": "bool", "default": False},
+        "synchronize_interface": {"type": "str", "required": True},
+        "synchronize_peer_ip": {"type": "str", "required": False},
+        "synchronize_config_to_ip": {"type": "str", "required": False},
+        "remote_system_username": {"type": "str", "required": False},
+        "remote_system_password": {"type": "str", "required": False, "no_log": True},
+        "services_to_synchronize": {
+            "type": "list",
+            "required": False,
+            "elements": "str",
+        },
+    }
```

Comment:
why arent' settings like `Disable preempt` and `Disconnect dialup interfaces` not supported?

Reply:
I orientated myself closely to the minimum viable product, which didn't include these settings. But there's nothing preventing me from adding them.

Reply:
7ecea2fd1500e9f558b4c566d336676a21487dff